python_modules/dagster/dagster/_core/definitions/time_window_partition_mapping.py

```diff
@@ -207,6 +223,45 @@ def _map_partitions(
 
                 time_windows.append(TimeWindow(window_start, window_end))
 
+        current_time = cast(
```

Comment:
nitpick, but shouldn't get_first_partition_window/get_last_partition_window handle this logic for us? i.e. if we pass in current_time = current_time (be it a real datetime or `None`), the partitions definition will generate a current time inside those method calls if necessary.

Reply:
oo yes i forgot about this